Docs: C API: Clarify what happens when null bytes are passed to `PyUnicode_AsUTF8` (#127458)
python/cpython: Doc/c-api/unicode.rst

   As :c:func:`PyUnicode_AsUTF8AndSize`, but does not store the size.

   .. note::

>       This function does not handle null bytes inside of *unicode*, so the length of the
>       returned string (from ``strlen()``) could be smaller than the length of the
>       passed unicode object, if the string contained embedded null characters. Prefer
>       :c:func:`PyUnicode_AsUTF8AndSize` when dealing with user input.

Reviewer: ```suggestion
      This function does not handle null bytes within the unicode object. As a result, the length of the
      returned string (from ``strlen()``) could be smaller than the length of the
      passed unicode object, if the string contained embedded null characters. When handling user input, 
      it is recommended to use :c:func:`PyUnicode_AsUTF8AndSize` instead.
```

   As :c:func:`PyUnicode_AsUTF8AndSize`, but does not store the size.

   .. note::

      This function does not handle null bytes within the unicode object. As a result, the length of the
      returned string (from ``strlen()``) could be smaller than the length of the

>       passed unicode object, if the string contained embedded null characters. When handling user input, 

Reviewer 2: Some extra trailing whitespace which is making the CI unhappy:
```suggestion
      passed unicode object, if the string contained embedded null characters. When handling user input,
```

   As :c:func:`PyUnicode_AsUTF8AndSize`, but does not store the size.

>    .. warning::

Reviewer 3: A warning is a strong signal. Maybe a note is enough?

Author: I originally had it as a note, but I think a warning is what we want here. I only discovered this quirk of `PyUnicode_AsUTF8` because it caused a crash in the `_interpreters` module. Things that can potentially cause security issues should probably get a warning, right?

Reviewer 3: You're right that notes about security are usually documented in red as **warnings**.

      This function does not have any special behavior for

>       `null bytes <https://en.wikipedia.org/wiki/Null_character>`_ embedded within
>       *unicode*. As a result, strings containing null bytes will remain in the returned

Reviewer 3: ```suggestion
      `null characters <https://en.wikipedia.org/wiki/Null_character>`_ embedded within
      *unicode*. As a result, strings containing null characters will remain in the returned
```

      string, which some C functions might interpret as the end of the string, leading to

>       truncation. When handling user input, it is recommended to use :c:func:`PyUnicode_AsUTF8AndSize`

Reviewer 3: I don't think that "user input" is useful here. I would prefer to say something like: "If truncation is an issue, ..."
